servatrice/src/servatrice_database_interface.cpp

```diff
@@ -1128,3 +1130,91 @@ int Servatrice_DatabaseInterface::checkNumberOfUserAccounts(const QString &email
 
     return 0;
 }
+
+bool Servatrice_DatabaseInterface::addForgotPassword(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	if (!updateUserToken(PasswordHasher::generateActivationToken(), user))
+		return false;
+
+	QSqlQuery *query = prepareQuery("insert into {prefix}_forgot_password (name,requestDate) values (:username,NOW())");
+	query->bindValue(":username", user);
+	if (execSqlQuery(query))
+		return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::removeForgotPassword(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	QSqlQuery *query = prepareQuery("delete from {prefix}_forgot_password where name = :username");
+	query->bindValue(":username", user);
+	if (execSqlQuery(query))
+		return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::doesForgotPasswordExist(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	QSqlQuery *query = prepareQuery("select count(name) from {prefix}_forgot_password where name = :user_name AND requestDate > (now() - interval :minutes minute)");
+	query->bindValue(":user_name", user);
+	query->bindValue(":minutes", QString::number(server->getForgotPasswordTokenLife()));
+
+	if (!execSqlQuery(query))
+		return false;
+
+	if (query->next())
+		if (query->value("count(name)").toInt() > 0)
+			return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::updateUserToken(const QString & token, const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	if (token.isEmpty() || user.isEmpty())
+		return false;
+
+	QSqlQuery *query = prepareQuery("update {prefix}_users set token = :token where name = :user_name");
```

Comment:
Currently the token field is used to store activation tokens generated for users when registering.  Once the account is active the code is no longer used any were else in the system.  Since the token is no longer needed after activation I see no issue with generating a new token, storing it here and sending it to the user for forgot password reasons.  Any reason you can think of?